src/sql/workbench/api/node/mainThreadConnectionManagement.ts

```diff
@@ -60,26 +60,31 @@ export class MainThreadConnectionManagement implements MainThreadConnectionManag
 	}
 
 	public async $openConnectionDialog(providers: string[], initialConnectionProfile?: IConnectionProfile, connectionCompletionOptions?: azdata.IConnectionCompletionOptions): Promise<azdata.connection.Connection> {
+		let connectionType = ConnectionType.editor;
```

Comment:
I'm not really sure what you're doing here?  the editor shouldn't be connecting the connection dialog through the extensibility APIs, should it?

Reply:
```TypeScript
let connectionType = ConnectionType.editor;
if (connectionCompletionOptions && !connectionCompletionOptions.saveConnection) {
	connectionType = ConnectionType.extension;
}
```